contributed_definitions/nyaml/NXfit.yaml

```diff
@@ -0,0 +1,263 @@
+category: base
+doc: |
+  Description of a fit procedure.
+symbols:
+  doc: |
+    The symbols used in the schema to specify e.g. dimensions of arrays.
+  n_points: |
+    Number of discretization points inside the fitting region.
+  n_independent: |
+    Number of independent axes for this fit procedure.
+type: group
+NXfit(NXobject):
+  label:
+    doc: |
+      Human-readable label for this fit procedure.
+  data(NXdata):
+    doc: |
+      Input data and results of the fit.
+    input_dependent(NX_NUMBER):
+      unit: NX_ANY
+      doc: |
+        Dependent axis for this fit procedure.
+      dimensions:
+        dim: [[1, n_points]]
+    input_independent(NX_NUMBER):
+      unit: NX_ANY
+      doc: |
+        Independent axis for this fit procedure.
+      dimensions:
+        dim: [[1, n_independent], [2, n_points]]
```

Comment:
Is this supposed to describe one dataset with possible multi independent coordinates (I(x, y, z)?) Then the dependent variable would not be an axis, but rather the data, no?

Reply:
It would be rather n_independent different n_points along the different dimensions, no? Like this I don't understand what it should represent

Reply:
Yeah, the idea was to represent multivariate data, i.e., `n_independent` variables, possibly each with a different number of  data points. I agree that it will not work like this because `n_points` is fixed for all variables like this. Any good ideas how to formulate this better?

Reply:
I aligned it a bit better (I think).

In **NXpeak/NXfit_background**:
```
symbols:
  n_points: |
    Number of discretization points at which the functional form of the peak
    has been evaluated.
  n_dependent: |
    Number of dependent variables to be fitted (for multidimensional/multivariate
    fit.)
NXpeak(NXobject):
  [...}
  data(NXdata):
    position(NX_NUMBER):
      unit: NX_ANY
      doc: |
        This axis holds the position values for the independent variable.
      dimensions:
        rank: 1
        dim: [[1, n_points]]
    intensity(NX_NUMBER):
      unit: NX_ANY
      doc: |
        This array holds the intensity/count values of the fitted peak at each position.
      dimensions:
        rank: 2
        dim: [[1, n_points], [2, n_dependent]]
```
In **NXfit**:
```
symbols:
  n_points: |
    Number of discretization points inside the fitting region.
  n_dependent: |
    Number of dependent variables to be fitted (for multidimensional/multivariate
    fit.)
NXfit(NXobject):
  [...]
  data(NXdata):
    doc: |
      Input data and results of the fit.
    input_dependent(NX_NUMBER):
      unit: NX_ANY
      doc: |
        Dependent axis for this fit procedure.
      dimensions:
        dim: [[1, n_points]]
    input_independent(NX_NUMBER):
      unit: NX_ANY
      doc: |
        Independent axis for this fit procedure.
      dimensions:
        rank: 2
        dim: [[1, n_points], [2, n_dependent]]
    envelope(NX_NUMBER):
      unit: NX_ANY
      doc: |
        Resulting envelope of all NXpeak instances.
        Should have the same dimension as the `position` axes of the individual peaks.
      dimensions:
        rank: 2
        dim: [[1, n_points], [2, n_dependent]]
    residual(NX_NUMBER):
      unit: NX_ANY
      doc: |
        Difference between the envelope and the data to be fitted.
      dimensions:
        rank: 2
        dim: [[1, n_points], [2, n_dependent]]
```

Like this, at least the symbols across the files match correctly. This still doesn't solve the probkem that each dependent variable can have different lengths.

Reply:
I used used a fixed `rank: dimRank` for all of these fields now, similarly to how it is done in [NXdata](https://fairmat-nfdi.github.io/nexus_definitions/classes/base_classes/NXdata.html#nxdata-data-field). The lengths of the individual axes cannot be preset and thus only in the docstring, it is said that the first axis of `input_dependent` must have the same number of points as the the first axis of `input_independent`.